firmware/src/sensors/ms5607.h

```diff
@@ -16,50 +16,158 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-/* MS5607 Barometer Device Library */
-
 #pragma once
 
-#include "cmsis_os.h"
+#include <cstdint>
+#include <vector>
+#include "drivers/gpio.h"
 #include "drivers/spi.h"
 
-/** Exported Types **/
+namespace sensors {
 
-enum ms5607_osr {
-  MS5607_OSR_256 = 0,
-  MS5607_OSR_512 = 1,
-  MS5607_OSR_1024 = 2,
-  MS5607_OSR_2048 = 3,
-  MS5607_OSR_4096 = 4,
-};
+class Ms5607 {
+ public:
+  /// Request measurement enum
+  enum class request : uint8_t {
+    PRESSURE = 0x40,
+    TEMPERATURE = 0x50,
+  };
 
-enum ms5607_data {
-  MS5607_PRESSURE = 1,
-  MS5607_TEMPERATURE = 2,
-};
+  /** Constructor
+   *
+   * @param spi reference the the SPI interface @injected
+   * @param cs reference the the chip select output pin @injected
+   */
+  Ms5607(drivers::Spi &spi, drivers::OutputPin &cs)
+      : m_spi(spi), m_cs(cs), m_last_request(request::PRESSURE), m_pressure{}, m_temperature{}, m_coefficients{} {
```

Comment:
curly braces initialization:

why: https://stackoverflow.com/questions/18222926/what-are-the-advantages-of-list-initialization-using-curly-braces